rust/src/measurements/exponential_max/mod.rs

```diff
@@ -0,0 +1,107 @@
+#[cfg(feature = "ffi")]
+mod ffi;
+
+use opendp_derive::bootstrap;
+
+use crate::{
+    core::{Function, Measurement, PrivacyMap},
+    domains::{AtomDomain, VectorDomain},
+    error::Fallible,
+    measures::MaxDivergence,
+    metrics::LInfDistance,
+    traits::{Float, InfAdd, InfCast, Number},
+};
+
+use crate::traits::{
+    samplers::{CastInternalRational, ExponentialPSRN},
+    DistanceConstant,
+};
+
+use super::{select_score, Optimize};
+
+#[cfg(test)]
+pub mod test;
+
+#[bootstrap(
+    features("contrib"),
+    arguments(optimize(c_type = "char *", rust_type = "String")),
+    generics(TIA(suppress))
+)]
+/// Make a Measurement that takes a vector of scores and privately selects the index of the highest score
+/// with noise added from the exponential distribution.
+///
+/// # Arguments
+/// * `input_domain` - Domain of the input vector. Must be a non-nullable VectorDomain.
+/// * `input_metric` - Metric on the input domain. Must be LInfDistance
+/// * `scale` - Higher scales are more private.
+/// * `optimize` - Indicate whether to privately return the "max" or "min"
+///
+/// # Generics
+/// * `TIA` - Atom Input Type. Type of each element in the score vector.
+/// * `QO` - Output Distance Type.
+pub fn make_report_noisy_max_exponential<TIA, QO>(
+    input_domain: VectorDomain<AtomDomain<TIA>>,
+    input_metric: LInfDistance<TIA>,
+    scale: QO,
+    optimize: Optimize,
+) -> Fallible<Measurement<VectorDomain<AtomDomain<TIA>>, usize, LInfDistance<TIA>, MaxDivergence<QO>>>
+where
+    TIA: Number + CastInternalRational,
+    QO: CastInternalRational + DistanceConstant<TIA> + Float,
+{
+    if input_domain.element_domain.nullable() {
+        return fallible!(MakeMeasurement, "values must be non-null");
+    }
+
+    if input_domain.element_domain.nullable() {
+        return fallible!(MakeMeasurement, "input domain must be non-nullable");
+    }
+
+    if scale.is_sign_negative() {
+        return fallible!(MakeMeasurement, "scale must not be negative");
+    }
+
+    let r_scale = scale.clone().into_rational()?;
+
+    Measurement::new(
+        input_domain,
+        Function::new_fallible(move |arg: &Vec<TIA>| {
+            select_score::<_, ExponentialPSRN>(
+                arg.iter().cloned(),
+                optimize.clone(),
+                r_scale.clone(),
+            )
+        }),
+        input_metric.clone(),
+        MaxDivergence::default(),
+        PrivacyMap::new_fallible(report_noisy_max_exponential_map(scale, input_metric)),
+    )
+}
+
+pub(crate) fn report_noisy_max_exponential_map<QI, QO>(
```

Comment:
This code is a duplicate of the gumbel map, however, I anticipate the gumbel map changing.